tests/Unit/IO/Test_H5.cpp

```diff
@@ -451,6 +459,88 @@ SPECTRE_TEST_CASE("Unit.IO.H5.DatRead", "[Unit][IO][H5]") {
   }
 }
 
+// Test that we can read scalar, rank-1, rank-2, and rank-3 datasets
+SPECTRE_TEST_CASE("Unit.IO.H5.ReadData", "[Unit][IO][H5]") {
+  const std::string h5_file_name("Unit.IO.H5.ReadData.h5");
+  const hid_t file_id =
+      H5Fcreate(h5_file_name.c_str(), H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT);
+  h5::detail::OpenGroup my_group(file_id, "ReadWrite",
+                                 h5::AccessType::ReadWrite);
+  const hid_t group_id = my_group.id();
+
+  using AllTypeList = tmpl::list<double, int, unsigned int, long, unsigned long,
+                                 long long, unsigned long long>;
+
+  const DataVector scalar_data{1.0};
+  const Index<0> scalar_extents{1};
+  h5::write_data(group_id, scalar_data, scalar_extents, "scalar_dataset");
+  tmpl::for_each<AllTypeList>([&group_id](auto x) noexcept {
+    typename decltype(x)::type scalar_data_from_file =
```

Comment:
you can save yourself typing by doing `using DataType = typename decltype(x)::type;` at the beginning of the lambda.